### PR TITLE
Update BitConverter from CoreFX

### PR DESCRIPTION
**Pull request by jkotas**
Related to #9701

### CODE REVIEW

**Comment by mikedn** [comment]
Would it make sense to copy the initialization of `IsLittleEndian` from corefx too? We'd avoid an `#if` and when  `IsLittleEndian` becomes an intrinsic the initialization cost won't be a concern anymore.

**Comment by jkotas** [comment]
I kept it because of there are number of BIGENDIAN ifdefs in corelib today, and so this is following the corelib style for endianness specific code.

**Comment by jkotas** [comment]
Wait for the current CoreCLR drop to stabilize in corefx before pushing this through (see https://github.com/dotnet/corefx/pull/16454)

**Comment by jkotas** [comment]
@stephentoub I see you have commented on the other PR related to this. Could you please take a look?

File: src/mscorlib/corefx/SR.cs [hunk 1]
```diff
@@ -815,4 +815,19 @@ internal static string Arg_UnauthorizedAccessException
     {
         get { return Environment.GetResourceString("Arg_UnauthorizedAccessException"); }
     }
+
+    internal static string ArgumentOutOfRange_GenericPositive
+    {
+        get { return Environment.GetResourceString("ArgumentOutOfRange_GenericPositive"); }
```
Comment (stephentoub) [inline src/mscorlib/corefx/SR.cs]: Nit: Not for this PR, but at some point subsequently we might want to go through and replace all of this strings with nameof, e.g.
```C#
using static Environment;
...
internal static string ArgumentOutOfRange_GenericPositive => GetResourceString(nameof(ArgumentOutOfRange_GenericPositive);
```

File: src/mscorlib/src/System/BitConverter.cs [hunk 2]
```diff
@@ -495,5 +452,20 @@ public static unsafe float Int32BitsToSingle(int value)
         {
             return *((float*)&value);
         }
+
+        private static void ThrowValueArgumentNull()
+        {
+            throw new ArgumentNullException("value");
```
Comment (stephentoub) [inline src/mscorlib/src/System/BitConverter.cs]: Did you purposefully decide against ThrowHelper for these exceptions?

File: src/mscorlib/corefx/SR.cs (hunk 1 above)
Comment (jkotas) [inline src/mscorlib/corefx/SR.cs]: The plan is switch this file to be auto-generated like everywhere else(https://github.com/dotnet/coreclr/issues/9474)

Comment (stephentoub) [inline src/mscorlib/corefx/SR.cs]: Even better.

File: src/mscorlib/src/System/BitConverter.cs (hunk 2 above)
Comment (jkotas) [inline src/mscorlib/src/System/BitConverter.cs]: Not really ... just copied what was in corefx. I have fixed it up to use ThrowHelper.

**Comment by hqueue** [comment]
@dotnet-bot test Linux ARM Emulator Cross Release Build
@dotnet-bot test Linux ARM Emulator Cross Debug Build

**Pull request merged**